src/ucp/api/ucp.h

```diff
@@ -2401,6 +2530,93 @@ ucs_status_ptr_t ucp_am_send_nb(ucp_ep_h ep, uint16_t id,
+ *       attributes by @ref ucp_worker_query routine.
+ *
+ *
+ * @return NULL                 - Active Message was sent immediately.
+ * @return UCS_PTR_IS_ERR(_ptr) - Error sending Active Message.
+ * @return otherwise            - Operation was scheduled for send and can be
+ *                                completed in any point in time. The request
+ *                                handle is returned to the application in order
+ *                                to track progress of the message. If user
+ *                                request was not provided in @a param->request,
+ *                                the application is responsible for releasing
+ *                                the handle using @ref ucp_request_free routine.
+ */
+ucs_status_ptr_t ucp_am_send_nbx(ucp_ep_h ep, uint16_t id,
+                                 const void *header, size_t header_length,
+                                 const void *buffer, size_t count,
+                                 ucp_request_param_t *param);
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Receive Active Message sent with rendezvous protocol.
+ *
+ * This routine receives a message that is described by the data descriptor
+ * @a data_desc, local address @a buffer, size @a count and @a param
+ * parameters on the @a worker. The routine is a non-blocking and therefore
+ * returns immediately. The receive operation is considered completed when the
+ * message is delivered to the @a buffer. If the receive operation cannot be
+ * stated the routine returns an error.
+ *
+ * @note After this call UCP takes an ownership of @a data_desc descriptor, so
+ *       there is no need to release it even if the operation fails.
+ *       The routine returns request handle instead, which can further be used
+ *       for tracking operation progress.
+ *
+ * @note This operation does not support CUDA memory.
+ *
+ * @param [in]  data_desc  Data descriptor, provided in
+                           @ref ucp_am_data_callback_t routine.
+ * @param [in]  buffer     Pointer to the buffer to receive the data to.
+ * @param [in]  count      Number of elements to receive into @a buffer.
+ * @param [in]  param      Operation parameters, see @ref ucp_request_param_t.
+ *
+ * @return UCS_PTR_IS_ERR(_ptr) - The receive operation failed.
+ * @return otherwise            - Operation was scheduled for send and can be
+ *                                completed in any point in time. The request
+ *                                handle is returned to the application in order
+ *                                to track progress of the message. If user
+ *                                request was not provided in @a param->request,
+ *                                the application is responsible for releasing
+ *                                the handle using @ref ucp_request_free routine.
+ */
+ucs_status_ptr_t ucp_am_recv_nbx(void *data_desc, void *buffer, size_t count,
```

Comment:
maybe ucp_am_recv_data_nbx()?

Reply:
ok

Reply:
renamed to `ucp_am_data_recv_nbx` to match others func names